Eslint fails on TypeScript components

<p><strong>Describe the bug</strong></p>
<p>eslint is misconfigured in some way causing it to fail if <code>react-native</code> is imported in a TypeScript file.</p>
<pre><code>/home/bill/Projects/Jellyfin/jellyfin-expo/screens/DownloadScreen.tsx
  13:45  error  Parse errors in imported module 'react-native': ';' expected. (14:32)  import/namespace
</code></pre>
<p>This appears to be a configuration issue of the import plugin in some way.</p>
<p>We should also probably be including <code>@types/react</code> and <code>@types/react-native</code> as dev dependencies for type definitions… unless those are already being included by Expo.</p>

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -32,6 +32,13 @@
         },
         "jest": {
             "version": "detect"
+        },
+        "import/resolver": {
+            "typescript": {
+                "alwaysTryTypes": true,
+                "project": "./tsconfig.json"
+            },
+            "node": true
         }
     },
     "rules": {
@@ -43,6 +50,8 @@
         "react-native/no-color-literals": ["error"],
         "react-native/no-raw-text": ["error"],
 
+        "import/no-named-as-default": "off",
+        "import/no-named-as-default-member": "off",
         "import/order": ["error", {
             "alphabetize": { "order": "asc", "caseInsensitive": true },
             "newlines-between": "always-and-inside-groups"
@@ -154,7 +163,12 @@
                 "plugin:import/warnings",
                 "plugin:import/typescript",
                 "plugin:eslint-comments/recommended"
-            ]
+            ],
+            "rules": {
+                "import/named": "off",
+                "import/namespace": "off",
+                "import/default": "off"
+            }
         }
     ]
 }

--- a/package.json
+++ b/package.json
@@ -100,6 +100,7 @@
     "babel-preset-expo": "~9.2.0",
     "domexception": "4.0.0",
     "eslint": "7.32.0",
+    "eslint-import-resolver-typescript": "4.4.4",
     "eslint-plugin-compat": "3.13.0",
     "eslint-plugin-eslint-comments": "3.2.0",
     "eslint-plugin-import": "2.31.0",

